wfl/Profiles/RestaurantCollectionProfile.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Hypermedia.Transforms;
+using Hypermedia;
+using Hypermedia.Affordances;
+using Symphono.Wfl.Controllers;
+using Symphono.Wfl.Models;
+using System;
+
+namespace Symphono.Wfl.Profiles
+{
+    public class RestaurantCollectionProfile : EntityProfile<RestaurantCollection>
+    {
+        protected override void Configure(EntityConfiguration<RestaurantCollection> configuration)
+        {
+            configuration
+                .UseRepresentationTransform(representation => representation
+                    .WithRepresentation("restaurant-collection")
+                    .WithRepresentation("collection")
+                )
+                .UseEnumerableEmbeddedSubEntityTransform(
+                    rc => rc.Restaurants,
+                    embedded => embedded
+                        .WithRelation("item")
+                        .WithConfiguration(embeddedConfiguration => embeddedConfiguration
+                            .UseProfile<Restaurant, RestaurantProfile>()
```

Comment:
So what the reflective profile function did was do a `dto.GetType()`, pulls in all the profiles for that `Type` and runs them. It also removes the double generic requirement here. The biggest value, is you no longer need to know what type of profile, it defaults to whatever the `dto.GetType()` is and uses that profile. It becomes more of a runtime concern instead of design time. You can still forcefully say, always use a specific profile like what's coded now.